thunder/core/jit_ext.py

```diff
@@ -895,7 +907,7 @@ def proxy_recursion(v):
             v.track_items()
             need_proxy = any(proxy_recursion(i) for i in v.item_wrappers)
         else:
-            need_proxy = isinstance(v.value, torch.Tensor)
+            need_proxy = isinstance(v.value, torch.Tensor) or isinstance(v.value, torch.device)
```

Comment:
I might be wrong, but could we just add the check but not proxy just like for `int` / `str`?

Reply:
It works (locally verified on the test). I have applied it in https://github.com/Lightning-AI/lightning-thunder/pull/556/commits/2073cb41384bcdae00c0baeaddbfbd95dd251ea7. Is this what you had in mind? Thanks!